app/plugin_install.go

```diff
@@ -333,21 +339,81 @@ func (a *App) removePlugin(id string) *model.AppError {
 	if err = a.removeSignature(id); err != nil {
 		mlog.Error("Can't remove signature", mlog.Err(err))
 	}
-
-	a.notifyClusterPluginEvent(
-		model.CLUSTER_EVENT_REMOVE_PLUGIN,
-		model.PluginEventData{
-			Id: id,
-		},
-	)
+	canaryTag := *a.Config().PluginSettings.CanaryTag
+	if canaryTag == "" {
```

Comment:
Could multiple servers in the cluster share a tag?

Reply:
yes they can.